lib/externalGradingSocket.sql

```diff
@@ -20,16 +20,11 @@ SELECT
     grading_job_status(gj.id) AS grading_job_status,
     s.variant_id AS variant_id
 FROM
-    grading_jobs AS gj
+    grading_jobs AS gj_orig
+    JOIN grading_jobs AS gj ON (gj.submission_id = gj_orig.submission_id)
```

Comment:
Checking my understanding here: we're finding the submissions associated with a particular grading job, then all the grading jobs associated with that submission, and then picking the most recent one?

Reply:
Yes. I'm just rewriting the same query that was there before to make it clearer. If the associated submission has a new grading job, the latest one is selected instead.